Test failure: com.linecorp.armeria.spring.ArmeriaAutoConfigurationWithoutMeterTest

Probably a flaky one:

```
com.linecorp.armeria.spring.ArmeriaAutoConfigurationWithoutMeterTest > test FAILED
    java.lang.AssertionError: 
    Expecting:
     <"# HELP armeria_executor_seconds_max  
    # TYPE armeria_executor_seconds_max gauge
    armeria_executor_seconds_max{name="blockingTaskExecutor",} 0.0
    # HELP armeria_executor_seconds  
    # TYPE armeria_executor_seconds summary
    armeria_executor_seconds_count{name="blockingTaskExecutor",} 0.0
    armeria_executor_seconds_sum{name="blockingTaskExecutor",} 0.0
    # HELP system_cpu_count The number of processors available to the Java virtual machine
    # TYPE system_cpu_count gauge
    system_cpu_count 8.0
    # HELP process_uptime_seconds The uptime of the Java virtual machine
    # TYPE process_uptime_seconds gauge
    process_uptime_seconds 22.36
    # HELP armeria_server_connections_lifespan_seconds_max  
    # TYPE armeria_server_connections_lifespan_seconds_max gauge
    armeria_server_connections_lifespan_seconds_max{protocol="h1c",} 0.0
    armeria_server_connections_lifespan_seconds_max{protocol="h2c",} 0.0
    # HELP armeria_server_connections_lifespan_seconds  
    # TYPE armeria_server_connections_lifespan_seconds summary
    armeria_server_connections_lifespan_seconds{protocol="h1c",quantile="0.0",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h1c",quantile="0.5",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h1c",quantile="0.75",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h1c",quantile="0.9",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h1c",quantile="0.95",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h1c",quantile="0.98",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h1c",quantile="0.99",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h1c",quantile="0.999",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h1c",quantile="1.0",} 0.0
    armeria_server_connections_lifespan_seconds_count{protocol="h1c",} 0.0
    armeria_server_connections_lifespan_seconds_sum{protocol="h1c",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h2c",quantile="0.0",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h2c",quantile="0.5",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h2c",quantile="0.75",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h2c",quantile="0.9",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h2c",quantile="0.95",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h2c",quantile="0.98",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h2c",quantile="0.99",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h2c",quantile="0.999",} 0.0
    armeria_server_connections_lifespan_seconds{protocol="h2c",quantile="1.0",} 0.0
    armeria_server_connections_lifespan_seconds_count{protocol="h2c",} 0.0
    armeria_server_connections_lifespan_seconds_sum{protocol="h2c",} 0.0
    # HELP armeria_executor_queue_remaining_tasks The number of additional elements that this queue can ideally accept without blocking
    # TYPE armeria_executor_queue_remaining_tasks gauge
    armeria_executor_queue_remaining_tasks{name="blockingTaskExecutor",} 2.147483647E9
    # HELP logback_events_total Number of error level events that made it to the logs
    # TYPE logback_events_total counter
    logback_events_total{level="warn",} 0.0
    logback_events_total{level="debug",} 16.0
    logback_events_total{level="error",} 0.0
    logback_events_total{level="trace",} 0.0
    logback_events_total{level="info",} 5.0
    # HELP armeria_server_router_virtual_host_cache_requests_total  
    # TYPE armeria_server_router_virtual_host_cache_requests_total counter
    armeria_server_router_virtual_host_cache_requests_total{hostname_pattern="*",result="miss",} 8.0
    armeria_server_router_virtual_host_cache_requests_total{hostname_pattern="*",result="hit",} 3.0
    # HELP process_start_time_seconds Start time of the process since unix epoch.
    # TYPE process_start_time_seconds gauge
    process_start_time_seconds 1.600527403195E9
    # HELP jvm_gc_live_data_size_bytes Size of old generation memory pool after a full GC
    # TYPE jvm_gc_live_data_size_bytes gauge
    jvm_gc_live_data_size_bytes 0.0
    # HELP process_cpu_usage The "recent cpu usage" for the Java Virtual Machine process
    # TYPE process_cpu_usage gauge
    process_cpu_usage 0.033761618370694364
    # HELP armeria_server_pending_responses  
    # TYPE armeria_server_pending_responses gauge
    armeria_server_pending_responses 1.0
    # HELP armeria_server_router_virtual_host_cache_eviction_weight_total  
    # TYPE armeria_server_router_virtual_host_cache_eviction_weight_total counter
    armeria_server_router_virtual_host_cache_eviction_weight_total{hostname_pattern="*",} 0.0
    # HELP armeria_server_connections  
    # TYPE armeria_server_connections gauge
    armeria_server_connections 2.0
    # HELP jvm_threads_peak_threads The peak live thread count since the Java virtual machine started or peak was reset
    # TYPE jvm_threads_peak_threads gauge
    jvm_threads_peak_threads 25.0
    # HELP armeria_server_requests_total  
    # TYPE armeria_server_requests_total counter
    armeria_server_requests_total{hostname_pattern="*",http_status="200",method="GET",result="success",service="okService",} 0.0
    armeria_server_requests_total{hostname_pattern="*",http_status="200",method="GET",result="failure",service="okService",} 0.0
    # HELP armeria_build_info A metric with a constant '1' value labeled by version and commit hash from which Armeria was built.
    # TYPE armeria_build_info gauge
    armeria_build_info{commit="0ec4de639cb10944f42d28f439cb8b674539a444",repo_status="clean",version="1.0.1-SNAPSHOT",} 1.0
    # HELP armeria_executor_pool_size_threads The current number of threads in the pool
    # TYPE armeria_executor_pool_size_threads gauge
    armeria_executor_pool_size_threads{name="blockingTaskExecutor",} 1.0
    # HELP jvm_memory_committed_bytes The amount of memory in bytes that is committed for the Java virtual machine to use
    # TYPE jvm_memory_committed_bytes gauge
    jvm_memory_committed_bytes{area="nonheap",id="CodeHeap 'profiled nmethods'",} 1.2386304E7
    jvm_memory_committed_bytes{area="heap",id="G1 Survivor Space",} 3145728.0
    jvm_memory_committed_bytes{area="heap",id="G1 Old Gen",} 5.24288E7
    jvm_memory_committed_bytes{area="nonheap",id="Metaspace",} 6.3778816E7
    jvm_memory_committed_bytes{area="nonheap",id="CodeHeap 'non-nmethods'",} 2555904.0
    jvm_memory_committed_bytes{area="heap",id="G1 Eden Space",} 2.5165824E7
    jvm_memory_committed_bytes{area="nonheap",id="Compressed Class Space",} 8990720.0
    jvm_memory_committed_bytes{area="nonheap",id="CodeHeap 'non-profiled nmethods'",} 3407872.0
    # HELP jvm_memory_max_bytes The maximum amount of memory in bytes that can be used for memory management
    # TYPE jvm_memory_max_bytes gauge
    jvm_memory_max_bytes{area="nonheap",id="CodeHeap 'profiled nmethods'",} 1.22908672E8
    jvm_memory_max_bytes{area="heap",id="G1 Survivor Space",} -1.0
    jvm_memory_max_bytes{area="heap",id="G1 Old Gen",} 5.36870912E8
    jvm_memory_max_bytes{area="nonheap",id="Metaspace",} -1.0
    jvm_memory_max_bytes{area="nonheap",id="CodeHeap 'non-nmethods'",} 5840896.0
    jvm_memory_max_bytes{area="heap",id="G1 Eden Space",} -1.0
    jvm_memory_max_bytes{area="nonheap",id="Compressed Class Space",} 1.073741824E9
    jvm_memory_max_bytes{area="nonheap",id="CodeHeap 'non-profiled nmethods'",} 1.22908672E8
    # HELP armeria_server_active_requests  
    # TYPE armeria_server_active_requests gauge
    armeria_server_active_requests{hostname_pattern="*",method="GET",service="okService",} 1.0
    # HELP jvm_buffer_memory_used_bytes An estimate of the memory that the Java virtual machine is using for this buffer pool
    # TYPE jvm_buffer_memory_used_bytes gauge
    jvm_buffer_memory_used_bytes{id="mapped",} 0.0
    jvm_buffer_memory_used_bytes{id="direct",} 8.4979999E7
    # HELP armeria_executor_completed_tasks_total The approximate total number of tasks that have completed execution
    # TYPE armeria_executor_completed_tasks_total counter
    armeria_executor_completed_tasks_total{name="blockingTaskExecutor",} 1.0
    # HELP armeria_executor_active_threads The approximate number of threads that are actively executing tasks
    # TYPE armeria_executor_active_threads gauge
    armeria_executor_active_threads{name="blockingTaskExecutor",} 0.0
    # HELP jvm_threads_states_threads The current number of threads having NEW state
    # TYPE jvm_threads_states_threads gauge
    jvm_threads_states_threads{state="runnable",} 13.0
    jvm_threads_states_threads{state="blocked",} 0.0
    jvm_threads_states_threads{state="waiting",} 7.0
    jvm_threads_states_threads{state="timed-waiting",} 5.0
    jvm_threads_states_threads{state="new",} 0.0
    jvm_threads_states_threads{state="terminated",} 0.0
    # HELP armeria_executor_scheduled_repetitively_total  
    # TYPE armeria_executor_scheduled_repetitively_total counter
    armeria_executor_scheduled_repetitively_total{name="blockingTaskExecutor",} 0.0
    # HELP armeria_server_file_vfs_cache_evictions_total  
    # TYPE armeria_server_file_vfs_cache_evictions_total counter
    armeria_server_file_vfs_cache_evictions_total{hostname_pattern="*",route="/internal/docs/*",vfs="DocService",} 0.0
    # HELP jvm_gc_memory_promoted_bytes_total Count of positive increases in the size of the old generation memory pool before GC to after GC
    # TYPE jvm_gc_memory_promoted_bytes_total counter
    jvm_gc_memory_promoted_bytes_total 1886208.0
    # HELP jvm_classes_loaded_classes The number of classes that are currently loaded in the Java virtual machine
    # TYPE jvm_classes_loaded_classes gauge
    jvm_classes_loaded_classes 12354.0
    # HELP armeria_client_connections_lifespan_seconds_max  
    # TYPE armeria_client_connections_lifespan_seconds_max gauge
    armeria_client_connections_lifespan_seconds_max{protocol="h1c",} 0.0
    armeria_client_connections_lifespan_seconds_max{protocol="h2c",} 0.0
    # HELP armeria_client_connections_lifespan_seconds  
    # TYPE armeria_client_connections_lifespan_seconds summary
    armeria_client_connections_lifespan_seconds{protocol="h1c",quantile="0.0",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h1c",quantile="0.5",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h1c",quantile="0.75",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h1c",quantile="0.9",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h1c",quantile="0.95",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h1c",quantile="0.98",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h1c",quantile="0.99",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h1c",quantile="0.999",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h1c",quantile="1.0",} 0.0
    armeria_client_connections_lifespan_seconds_count{protocol="h1c",} 0.0
    armeria_client_connections_lifespan_seconds_sum{protocol="h1c",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h2c",quantile="0.0",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h2c",quantile="0.5",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h2c",quantile="0.75",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h2c",quantile="0.9",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h2c",quantile="0.95",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h2c",quantile="0.98",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h2c",quantile="0.99",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h2c",quantile="0.999",} 0.0
    armeria_client_connections_lifespan_seconds{protocol="h2c",quantile="1.0",} 0.0
    armeria_client_connections_lifespan_seconds_count{protocol="h2c",} 0.0
    armeria_client_connections_lifespan_seconds_sum{protocol="h2c",} 0.0
    # HELP jvm_threads_daemon_threads The current number of live daemon threads
    # TYPE jvm_threads_daemon_threads gauge
    jvm_threads_daemon_threads 17.0
    # HELP process_files_open_files The open file descriptor count
    # TYPE process_files_open_files gauge
    process_files_open_files 233.0
    # HELP jvm_buffer_total_capacity_bytes An estimate of the total capacity of the buffers in this pool
    # TYPE jvm_buffer_total_capacity_bytes gauge
    jvm_buffer_total_capacity_bytes{id="mapped",} 0.0
    jvm_buffer_total_capacity_bytes{id="direct",} 8.4979998E7
    # HELP armeria_server_router_virtual_host_cache_evictions_total  
    # TYPE armeria_server_router_virtual_host_cache_evictions_total counter
    armeria_server_router_virtual_host_cache_evictions_total{hostname_pattern="*",} 0.0
    # HELP jvm_buffer_count_buffers An estimate of the number of buffers in the pool
    # TYPE jvm_buffer_count_buffers gauge
    jvm_buffer_count_buffers{id="mapped",} 0.0
    jvm_buffer_count_buffers{id="direct",} 39.0
    # HELP jvm_classes_unloaded_classes_total The total number of classes unloaded since the Java virtual machine has started execution
    # TYPE jvm_classes_unloaded_classes_total counter
    jvm_classes_unloaded_classes_total 1.0
    # HELP jvm_gc_pause_seconds Time spent in GC pause
    # TYPE jvm_gc_pause_seconds summary
    jvm_gc_pause_seconds_count{action="end of minor GC",cause="G1 Evacuation Pause",} 1.0
    jvm_gc_pause_seconds_sum{action="end of minor GC",cause="G1 Evacuation Pause",} 0.013
    # HELP jvm_gc_pause_seconds_max Time spent in GC pause
    # TYPE jvm_gc_pause_seconds_max gauge
    jvm_gc_pause_seconds_max{action="end of minor GC",cause="G1 Evacuation Pause",} 0.013
    # HELP jvm_gc_max_data_size_bytes Max size of old generation memory pool
    # TYPE jvm_gc_max_data_size_bytes gauge
    jvm_gc_max_data_size_bytes 5.36870912E8
    # HELP jvm_threads_live_threads The current number of live threads including both daemon and non-daemon threads
    # TYPE jvm_threads_live_threads gauge
    jvm_threads_live_threads 25.0
    # HELP system_load_average_1m The sum of the number of runnable entities queued to available processors and the number of runnable entities running on the available processors averaged over a period of time
    # TYPE system_load_average_1m gauge
    system_load_average_1m 15.98
    # HELP system_cpu_usage The "recent cpu usage" for the whole system
    # TYPE system_cpu_usage gauge
    system_cpu_usage 0.9769754768392371
    # HELP armeria_server_file_vfs_cache_eviction_weight_total  
    # TYPE armeria_server_file_vfs_cache_eviction_weight_total counter
    armeria_server_file_vfs_cache_eviction_weight_total{hostname_pattern="*",route="/internal/docs/*",vfs="DocService",} 0.0
    # HELP armeria_server_router_virtual_host_cache_estimated_size  
    # TYPE armeria_server_router_virtual_host_cache_estimated_size gauge
    armeria_server_router_virtual_host_cache_estimated_size{hostname_pattern="*",} 8.0
    # HELP process_files_max_files The maximum file descriptor count
    # TYPE process_files_max_files gauge
    process_files_max_files 1048576.0
    # HELP armeria_server_file_vfs_cache_requests_total  
    # TYPE armeria_server_file_vfs_cache_requests_total counter
    armeria_server_file_vfs_cache_requests_total{hostname_pattern="*",result="miss",route="/internal/docs/*",vfs="DocService",} 0.0
    armeria_server_file_vfs_cache_requests_total{hostname_pattern="*",result="hit",route="/internal/docs/*",vfs="DocService",} 0.0
    # HELP armeria_executor_pool_core_threads The core number of threads for the pool
    # TYPE armeria_executor_pool_core_threads gauge
    armeria_executor_pool_core_threads{name="blockingTaskExecutor",} 200.0
    # HELP armeria_executor_pool_max_threads The maximum allowed number of threads in the pool
    # TYPE armeria_executor_pool_max_threads gauge
    armeria_executor_pool_max_threads{name="blockingTaskExecutor",} 2.147483647E9
    # HELP armeria_server_file_vfs_cache_estimated_size  
    # TYPE armeria_server_file_vfs_cache_estimated_size gauge
    armeria_server_file_vfs_cache_estimated_size{hostname_pattern="*",route="/internal/docs/*",vfs="DocService",} 0.0
    # HELP jvm_gc_memory_allocated_bytes_total Incremented for an increase in the size of the young generation memory pool after one GC to before the next
    # TYPE jvm_gc_memory_allocated_bytes_total counter
    jvm_gc_memory_allocated_bytes_total 2.3068672E7
    # HELP armeria_executor_scheduled_once_total  
    # TYPE armeria_executor_scheduled_once_total counter
    armeria_executor_scheduled_once_total{name="blockingTaskExecutor",} 0.0
    # HELP armeria_executor_queued_tasks The approximate number of tasks that are queued for execution
    # TYPE armeria_executor_queued_tasks gauge
    armeria_executor_queued_tasks{name="blockingTaskExecutor",} 0.0
    # HELP jvm_memory_used_bytes The amount of used memory
    # TYPE jvm_memory_used_bytes gauge
    jvm_memory_used_bytes{area="nonheap",id="CodeHeap 'profiled nmethods'",} 1.2384896E7
    jvm_memory_used_bytes{area="heap",id="G1 Survivor Space",} 2474080.0
    jvm_memory_used_bytes{area="heap",id="G1 Old Gen",} 3.4127872E7
    jvm_memory_used_bytes{area="nonheap",id="Metaspace",} 6.1616848E7
    jvm_memory_used_bytes{area="nonheap",id="CodeHeap 'non-nmethods'",} 1423232.0
    jvm_memory_used_bytes{area="heap",id="G1 Eden Space",} 1.3631488E7
    jvm_memory_used_bytes{area="nonheap",id="Compressed Class Space",} 8169880.0
    jvm_memory_used_bytes{area="nonheap",id="CodeHeap 'non-profiled nmethods'",} 3365504.0
    # HELP armeria_executor_idle_seconds  
    # TYPE armeria_executor_idle_seconds summary
    armeria_executor_idle_seconds_count{name="blockingTaskExecutor",} 0.0
    armeria_executor_idle_seconds_sum{name="blockingTaskExecutor",} 0.0
    # HELP armeria_executor_idle_seconds_max  
    # TYPE armeria_executor_idle_seconds_max gauge
    armeria_executor_idle_seconds_max{name="blockingTaskExecutor",} 0.0
    ">
    to contain:
     <"# TYPE armeria_server_response_duration_seconds_max gauge"> 
        at com.linecorp.armeria.spring.ArmeriaAutoConfigurationWithoutMeterTest.test(ArmeriaAutoConfigurationWithoutMeterTest.java:85)
```

--- a/spring/boot2-autoconfigure/src/test/java/com/linecorp/armeria/spring/ArmeriaAutoConfigurationWithoutMeterTest.java
+++ b/spring/boot2-autoconfigure/src/test/java/com/linecorp/armeria/spring/ArmeriaAutoConfigurationWithoutMeterTest.java
@@ -29,13 +29,10 @@ import org.junit.rules.Timeout;
 import org.junit.runner.RunWith;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.context.annotation.Import;
 import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.junit4.SpringRunner;
 
 import com.linecorp.armeria.client.WebClient;
-import com.linecorp.armeria.common.AggregatedHttpResponse;
-import com.linecorp.armeria.common.HttpStatus;
 import com.linecorp.armeria.common.metric.MeterIdPrefixFunction;
 import com.linecorp.armeria.server.Server;
 import com.linecorp.armeria.spring.ArmeriaAutoConfigurationWithoutMeterTest.NoMeterTestConfiguration;
@@ -55,7 +52,6 @@ public class ArmeriaAutoConfigurationWithoutMeterTest {
      * {@link MeterIdPrefixFunction} and {@link MeterRegistry} are not registered as bean.
      */
     @SpringBootApplication
-    @Import(ArmeriaOkServiceConfiguration.class)
     public static class NoMeterTestConfiguration {
     }
 
@@ -72,16 +68,10 @@ public class ArmeriaAutoConfigurationWithoutMeterTest {
 
     @Test
     public void test() {
-        final AggregatedHttpResponse msg = WebClient.of(newUrl("h1c"))
-                                                    .get("/ok")
-                                                    .aggregate().join();
-        assertThat(msg.status()).isEqualTo(HttpStatus.OK);
-        assertThat(msg.contentUtf8()).isEqualTo("ok");
-
         final String metricReport = WebClient.of(newUrl("http"))
                                              .get("/internal/metrics")
                                              .aggregate().join()
                                              .contentUtf8();
-        assertThat(metricReport).contains("# TYPE armeria_server_response_duration_seconds_max gauge");
+        assertThat(metricReport).contains("# TYPE armeria_server_connections gauge");
     }
 }

--- a/spring/boot2-autoconfigure/src/test/java/com/linecorp/armeria/spring/ArmeriaAutoConfigurationWithoutMeterTest.java
+++ b/spring/boot2-autoconfigure/src/test/java/com/linecorp/armeria/spring/ArmeriaAutoConfigurationWithoutMeterTest.java
@@ -16,6 +16,7 @@
 package com.linecorp.armeria.spring;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
 
 import java.util.concurrent.TimeUnit;
 
@@ -68,10 +69,12 @@ public class ArmeriaAutoConfigurationWithoutMeterTest {
 
     @Test
     public void test() {
-        final String metricReport = WebClient.of(newUrl("http"))
-                                             .get("/internal/metrics")
-                                             .aggregate().join()
-                                             .contentUtf8();
-        assertThat(metricReport).contains("# TYPE armeria_server_connections gauge");
+        await().untilAsserted(() -> {
+            final String metricReport = WebClient.of(newUrl("h1c"))
+                                                 .get("/internal/metrics")
+                                                 .aggregate().join()
+                                                 .contentUtf8();
+            assertThat(metricReport).contains("# TYPE armeria_server_connections gauge");
+        });
     }
 }

